_squirtle/mod.ts

```diff
@@ -437,45 +480,115 @@ const runRPC = async (argv: any) => {
+    case 'catching_up':
+      shortStatus = '⏳'
+      break
+    case 'ready':
+      shortStatus = '📡'
+      break
+  }
+  console.log(`(🧮 local ${shortStatus})`)
+}
+
 const runHelp = async () => {
   const run1 = Deno.run({
     cmd: ['sq', 'help'],
   })
   await run1.status()
 }
 
+const gp = (opts: Deno.RunOptions): Promise<string> => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const cmd: readonly string[] = ['gp', ...(opts.cmd as string[] ?? [])]
+      const runOpts = {
+        cmd,
+        cwd: opts.cwd,
+        stdout: opts.stdout ?? 'piped',
+        stderr: opts.stderr ?? 'piped',
+      }
+      const gpProcess = Deno.run(runOpts)
+      const [status, stdout, stderr] = await Promise.all([
+        gpProcess.status(),
+        runOpts.stdout === 'piped' ? gpProcess.output() : undefined,
+        runOpts.stderr === 'piped' ? gpProcess.stderrOutput() : undefined,
+      ])
+      gpProcess.close();
+
+      if (status.success) {
+        resolve(new TextDecoder().decode(stdout).trim())
+      } else {
+        reject(new TextDecoder().decode(stderr))
+      }
+    } catch (e) {
+      if (e == 'NotFound: No such file or directory (os error 2)') {
+        reject("command 'gp' is only available in a gitpod")
+      } reject(e)
+    }
+  })
+}
+
+const getDotFilesLocation = async () => {
+  const url = await gp({ cmd: ['url'] })
+    .catch(() => false)
+
+  if (typeof url === 'string' && url.indexOf('gitpod.io') !== -1) {
+    return '/workspace'
+  }
+
+  return await getRootDir()
+}
+
 const getRootDir = async () => {
-  const rootDir = String.fromCharCode(...await Deno.run({
+  return await Deno.run({
     cmd: ['git', 'rev-parse', '--show-toplevel'],
     stdout: "piped",
-  }).output()).replace("\n", "")
-  return await Deno.realPath(rootDir)
+  })
+    .output()
+    .then((path: BufferSource) => new TextDecoder().decode(path).trim())
+    .then((path: string) => Deno.realPathSync(path))
 }
 
 const openLatestReadme = async () => {
   const rootDir = await getRootDir()
-  
-  const latestReadmeRun = Deno.run({
+
+  const latestReadme = await Deno.run({
     cwd: rootDir,
     cmd: ['find', 'quests', '-name', 'README.md'],
     stdout: "piped",
-    stderr: "piped"
   })
-  await latestReadmeRun.status()
+    .output()
+    .then((output: BufferSource) => new TextDecoder().decode(output)
+      .trim()
+      .split("\n")
+      .pop() ?? '')
 
-  const readme = String.fromCharCode(... await latestReadmeRun.output()).split("\n").filter(s => s !== "").reverse()[0]
-  await Deno.run({
+  await gp({
     cwd: rootDir,
-    cmd: ['gp', 'open', readme]
-  }).status()
+    cmd: ['open', latestReadme],
+    stdout: 'null',
+    stderr: 'null',
+  }).catch(() => { })
 }
 
 const getEnv = async () => {
-  const run1 = Deno.run({
-    cmd: ['gp', 'env'],
-    stdout: 'piped'
-  })
-  const gpEnvString = new TextDecoder().decode(await run1.output()).trim()
+  const gpEnvString = await gp({ cmd: ['env'] })
```

Comment:
This should still be piped right otherwise we get no env string?

Reply:
`gp` is always doing the pipe internally by default

Reply:
👍 as long as you're confident and it's tested I'm cool with it